browser/brave_wallet/keyring_service_unittest.cc

```diff
@@ -4432,4 +4435,58 @@ TEST_F(KeyringServiceUnitTest, DevWalletPassword) {
 }
 #endif  // !defined(OFFICIAL_BUILD)
 
+TEST_F(KeyringServiceUnitTest, BitcoinReceiveChangeAddresses) {
+  // TODO(apaymyshev): update existing tests above to also cover Bitcoin
+  // keyring.
+
+  base::test::ScopedFeatureList feature_list{
+      features::kBraveWalletBitcoinFeature};
+
+  KeyringService service(json_rpc_service(), GetPrefs(), GetLocalState());
+
+  // https://github.com/bitcoin/bips/blob/master/bip-0084.mediawiki#test-vectors
+  ASSERT_TRUE(RestoreWallet(
+      &service,
+      "abandon abandon abandon abandon abandon abandon abandon abandon abandon "
+      "abandon abandon about",
+      "brave", false));
+
+  EXPECT_THAT(
+      service.GetKeyringInfoSync(mojom::kBitcoinKeyringId)->account_infos,
+      testing::IsEmpty());
+
+  EXPECT_EQ(service.GetBitcoinReceivingAddress(mojom::kBitcoinKeyringId, 0, 0),
+            absl::nullopt);
+  EXPECT_EQ(service.GetBitcoinChangeAddress(mojom::kBitcoinKeyringId, 0, 0),
+            absl::nullopt);
+  EXPECT_EQ(service.GetBitcoinReceivingAddress(mojom::kBitcoinKeyringId, 1, 0),
+            absl::nullopt);
+  EXPECT_EQ(service.GetBitcoinChangeAddress(mojom::kBitcoinKeyringId, 1, 0),
+            absl::nullopt);
+
+  AddAccount(&service, "Btc Acc", mojom::CoinType::BTC);
+
+  EXPECT_THAT(
+      service.GetKeyringInfoSync(mojom::kBitcoinKeyringId)->account_infos,
+      testing::SizeIs(1u));
+  auto btc_acc = service.GetKeyringInfoSync(mojom::kBitcoinKeyringId)
+                     ->account_infos[0]
+                     ->Clone();
+  EXPECT_EQ(btc_acc->address, "bc1ql5f64jdzjsvgehlpxvdgm9ygp0xta7xpnueh03");
```

Comment:
Is having an address for the account necessary? Perhaps `btc_acc->xpub` makes more sense here.

What also makes sense (for a future PR) is to have a `btc_acc->GetFreshReceivingAddress()` which returns an address for the first receiving index that is not used already (i.e., has no inputs/outputs on it). This is the address that we might want to expose to the user for deposits.

Reply:
Yep, account address does not make sense for bitcoin. But we now have 1-to-1 Account<->Address relation invariant which should be somehow handled in future. I just cover that behavior in tests.

`btc_acc->GetFreshReceivingAddress()` - sure, something like that was planned when RPC stuff is ready :) Thanks!

Reply:
We can make the address in the `AccountInfo` a union just like `TxDataUnion` so each different coin can encapsulate their own specific data.